### PR TITLE
cut down EACAP gas usage when access control is disabled

### PR DESCRIPTION
**Pull request by se3000**

### CODE REVIEW

File: evm-contracts/src/v0.6/EACAggregatorProxy.sol [hunk 1]
```diff
@@ -254,7 +254,9 @@ contract EACAggregatorProxy is AggregatorProxy {
    * contract or is the contract itself.
    */
   modifier checkAccess() {
-    require(accessController.hasAccess(msg.sender, msg.data), "No access");
+    AccessControllerInterface ac = accessController;
+    bool disabled = address(ac) == address(0);
+    require(disabled || ac.hasAccess(msg.sender, msg.data), "No access");
```
Comment (thodges-gh) [inline evm-contracts/src/v0.6/EACAggregatorProxy.sol:259]: You can save 5 gas by removing the line above:
```suggestion
    require(address(ac) == address(0) || ac.hasAccess(msg.sender, msg.data), "No access");
```
:joy:

Comment (se3000) [reply evm-contracts/src/v0.6/EACAggregatorProxy.sol]: Adds up.

File: evm-contracts/src/v0.6/EACAggregatorProxy.sol [hunk 2]
```diff
@@ -254,7 +254,9 @@ contract EACAggregatorProxy is AggregatorProxy {
    * contract or is the contract itself.
    */
   modifier checkAccess() {
-    require(accessController.hasAccess(msg.sender, msg.data), "No access");
+    AccessControllerInterface ac = accessController;
+    bool disabled = address(ac) == address(0);
+    require(address(ac) == address(0) || ac.hasAccess(msg.sender, msg.data), "No access");
```
Comment (krebernisak) [inline evm-contracts/src/v0.6/EACAggregatorProxy.sol:259]:
```suggestion
    require(disabled || ac.hasAccess(msg.sender, msg.data), "No access");
```

File: evm-contracts/src/v0.6/EACAggregatorProxy.sol [hunk 3]
```diff
@@ -254,7 +254,9 @@ contract EACAggregatorProxy is AggregatorProxy {
    * contract or is the contract itself.
    */
   modifier checkAccess() {
-    require(accessController.hasAccess(msg.sender, msg.data), "No access");
+    AccessControllerInterface ac = accessController;
+    bool disabled = address(ac) == address(0);
```
Comment (krebernisak) [inline evm-contracts/src/v0.6/EACAggregatorProxy.sol:258]:
```suggestion
```

File: evm-contracts/test/v0.6/EACAggregatorProxy.test.ts [hunk 4]
```diff
@@ -206,6 +210,19 @@ describe('EACAggregatorProxy', () => {
         }, 'No proposed aggregator present')
       })
     })
+
+    describe('when read from a contract that is not permissioned', () => {
+      beforeEach(async () => {
+        testHelper = await testHelperFactory.connect(personas.Carol).deploy()
```
Comment (topliceanu) [inline evm-contracts/test/v0.6/EACAggregatorProxy.test.ts:216]: I'm not sure how this works, but it seems an `afterEach()` where you `"disconnect"` from the contract would be help.

File: evm-contracts/src/v0.6/EACAggregatorProxy.sol (hunk 3 above)
Comment (kaleofduty) [reply evm-contracts/src/v0.6/EACAggregatorProxy.sol:258]: nowhere, hence @krebernisak 's suggestion I'm guessing

File: evm-contracts/test/v0.6/EACAggregatorProxy.test.ts (hunk 4 above)
Comment (kaleofduty) [reply evm-contracts/test/v0.6/EACAggregatorProxy.test.ts:216]: the `connect` name is a confusing. No persistent connection or similar is created, so there's no need to clean anything up

**Pull request merged**